### ISSUE TITLE
Investigate other DDrawCompat hacks to fix sporadic slowdowns

### ISSUE THREAD

**Issue by nmlgc** [nmlgc/ssg#35]
hile the 32-bit mode implemented in #3, it seems to have introduced (or, not prevented) sporadic slowdowns at very specific places, for example the defeat animation of the Extra Stage midboss. This happened to @WishMakers0 on stream:

https://clips.twitch.tv/CrowdedFlaccidAppleTBCheesePull-CWOmyHfFm1mcmZSf

We might need other DDrawCompat hacks after all.

**Comment by nmlgc** [comment]
OK, I'm still getting this slowdown with SDL_Renderer's Direct3D 12 backend on my system. However, it also confirms that the cause is indeed an overwhelming number of draw calls; setting the environment variable `SDL_RENDER_BATCHING` to `1` works around the issue.\
The other backends work fine even without batching.

**Comment by WishMakers0** [comment]
That checks out.  Running other games seems to imply that bullet cancels are a common optimization problem with older forms of DirectX in shmups.  It's a sheer numbers game seemingly no matter whether it's just text or "particle" animations for bullet destruction, I noticed it in a couple other games too.
Batching seems like a reasonable solution to the problem without changing very much underlying functionality in other places 👍

**Comment by nmlgc** [comment]
Yes, it's very much related to a lack of batching. And not just Direct3D 12; I just saw the Direct3D 9 backend drop to 33 FPS on a 13-year old mid-range laptop (Intel Core i5-2520M, Intel HD Graphics 3000) under Wine. The OpenGL backend merely dropped to 63 FPS, from a median of 176 FPS over the full Extra Stage.
